Cfast node error fix for ccf

mm-broadcom opened the pull request:
## How Has This Been Tested?

Ran the LSP locally while having the extension point to it. Replicated steps that caused the bug prior to the fix and the nodes worked as intended.

## Checklist:
- [x] Each of my commits contains one meaningful change
- [x] I have performed rebase of my branch on top of the development
- [x] I have performed a self-review of my own code
- [] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I have checked my code and corrected any misspellings

mm-broadcom committed:
Initial commit

Initial commit - Has not fixed the issue. Committing due to GitHub issue when trying to change branches.

mm-broadcom committed:
Remove duplicate imports

mm-broadcom committed:
Properly form the URI before using

mm-broadcom committed:
Provide Path object directly

mm-broadcom committed:
Removed unused import

mm-broadcom committed:
Get copybook text via CopybookService

mm-broadcom committed:
Remove unused code

mm-broadcom committed:
Load copybook as List<Line>

mm-broadcom committed:
Fix checkstyle

mm-broadcom committed:
Ran formatter

ishche commented on server/engine/src/test/java/org/eclipse/lsp/cobol/extendedapi/CFASTBuilderTest.java:
```diff
@@ -57,9 +67,10 @@ void cfastBuilderTest(String src, String jsonTree, String caseName) {
     AnalysisResult analysisResult =
         UseCaseUtils.analyze(UseCase.builder().documentUri("fake/path").text(src).build());
     DocumentModelService documentModelService = new DocumentModelService();
+    CopybookServiceImpl copybookService = createCopybookService();
```
will it be enough just to mock it?

mm-broadcom replied:
It should be. The service is already being put through the wringer in its own dedicated test class: https://github.com/eclipse-che4z/che-che4z-lsp-for-cobol/blob/development/server/engine/src/test/java/org/eclipse/lsp/cobol/service/copybooks/CopybookServiceTest.java

I just borrowed the function to create the mock to use here such that it satisfies the constructor.

ishche commented on server/engine/src/main/java/org/eclipse/lsp/cobol/cfg/CFASTBuilderImpl.java:
```diff
@@ -286,16 +290,41 @@ private Location convertLocation(Node node) {
+        if (nodeURIString.equals(copybook.getUri())) {
+          String[] tempArr = copybook.getContent().split("\\r?\\n", -1);
+          for (int i = 0; i < tempArr.length; i++) {
+            resultText.add(new CobolDocumentModel.Line(i, tempArr[i]));
```
Is it possible to use doc.getText() instead of doc.getLines to avoid objects creation?

mm-broadcom replied:
I tried that with [an earlier commit](https://github.com/eclipse-che4z/che-che4z-lsp-for-cobol/pull/2804/commits/4fa8888ae66ac92883fa5600a1532c14fdb9ab84) and because other things do `var.toString()` rather than grab the text, it broke a lot of the tests for CFAST.

slavek-kucera replied:
I agree that this needs to be handled differently. The split might be performed for every single node.
I also I don't see the reason for creating the `CobolDocumentModel.Line` objects, just append the useful lines.

slavek-kucera commented on server/engine/src/main/java/org/eclipse/lsp/cobol/cfg/CFASTBuilderImpl.java:
```diff
@@ -42,10 +43,13 @@
   @Inject
-  public CFASTBuilderImpl(DocumentModelService documentModelService) {
+  public CFASTBuilderImpl(
+      DocumentModelService documentModelService, CopybookServiceImpl copybookService) {
```
We should not depend on the implementation here.

slavek-kucera commented on server/engine/src/main/java/org/eclipse/lsp/cobol/cfg/CFASTBuilderImpl.java:
```diff
@@ -286,16 +290,41 @@ private Location convertLocation(Node node) {
+        return "<snippet creation error>";
+      }
+
+      Set<CopybookModel> copybookSet = copybookService.getCopybookUsage(rootURIString);
```
This can be performed just once for the whole source, right?

mm-broadcom committed:
Use CopybookService interface for injection

mm-broadcom committed:
Ran formatter

mm-broadcom committed:
Remove errant logs, only parse what is needed of the copybooks

mm-broadcom committed:
Use unprocessed content for copybook snippets

Store the original, unprocessed, content within CopybookModel for use where comments and other removed details are required. Changed the content used for the snippets to this original one as well.

mm-broadcom committed:
Ran formatter

mm-broadcom committed:
Get CopybookSet once per program versus per-node
